Code/GraphMol/FileParsers/file_parsers_catch.cpp

```diff
@@ -5511,4 +5515,181 @@ NO_CHARGES
     std::unique_ptr<RWMol> m(Mol2BlockToMol(mol2));
     REQUIRE(m);
   }
-}
\ No newline at end of file
+}
+
+std::string read_file(const std::string &fname) {
+  std::ifstream ifs(fname);
+  return std::string(std::istreambuf_iterator<char>(ifs),
+                     std::istreambuf_iterator<char>());
+}
+
+#ifdef RDK_BUILD_MAEPARSER_SUPPORT
+TEST_CASE("MaeMolSupplier setData and reset methods", "[MaeMolSupplier]") {
+  std::string rdbase = getenv("RDBASE");
+
+  MaeMolSupplier supplier;
+
+  std::vector<std::string> mol_names1 = {
+      "48",  "78",  "128", "163", "164", "170", "180", "186",
+      "192", "203", "210", "211", "213", "220", "229", "256"};
+  std::string fname1 =
+      rdbase + "/Code/GraphMol/FileParsers/test_data/NCI_aids_few.mae";
+  auto data1 = read_file(fname1);
+
+  supplier.setData(data1);
+
+  std::unique_ptr<ROMol> mol = nullptr;
```

Comment:
I don't think `mol` needs to be defined outside the loop

Reply:
Yup.